src/main/java/com/google/api/pathtemplate/PathTemplate.java

```diff
@@ -1015,6 +1013,20 @@ public String encode(String... values) {
     return builder.build();
   }
 
+  private static boolean isSegmentBeginOrEndInvalid(String seg) {
+    // A segment is invalid if it contains only one character and the character is a delimiter
+    if (seg.length() == 1 && COMPLEX_DELIMITER_PATTERN.matcher(seg).find()) {
+      return true;
+    }
+    // A segment can start with a delimiter, as long as there is no { right after it.
+    // A segment can end with a delimiter, as long as there is no } right before it.
+    // e.g. Invalid: .{well}-{known}, {well}-{known}-
+    // Valid: .well-known, .well-{known}, .-~{well-known}, these segments are all considered as literals for matching
+    return (COMPLEX_DELIMITER_PATTERN.matcher(seg.substring(0, 1)).find() && seg.charAt(1) == '{')
```

Comment:
Just to confirm: what if the segment is two consecutive delimiters (e.g.,, `..` of length 2)? According to this check, it's valid. Perhaps this is checked in other places?

Also, `well.{known}.stuff` is valid, right?

Reply:
yes, 2+ consecutive delimiters in between curly brackets are checked [here](https://github.com/googleapis/api-common-java/blob/6a8a0a44fbf942ea6224ab3d8d4b3bcc6c79c5b4/src/main/java/com/google/api/pathtemplate/PathTemplate.java#L896), it's valid if they are in the beginning since they are considered literals.  
`well.{known}.stuff` is also considered a literal hence is valid.